providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/model/FlagParser.java

```diff
@@ -40,37 +39,39 @@ private FlagParser() {
     }
 
     static {
-        try (InputStream schema = FlagParser.class.getClassLoader().getResourceAsStream(SCHEMA_RESOURCE)) {
-            if (schema == null) {
-                log.warn(String.format("Resource %s not found", SCHEMA_RESOURCE));
-            } else {
-                final ByteArrayOutputStream result = new ByteArrayOutputStream();
-                byte[] buffer = new byte[512];
-                for (int size; 0 < (size = schema.read(buffer)); ) {
-                    result.write(buffer, 0, size);
-                }
-
-                JsonSchemaFactory instance = JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V7);
-                SCHEMA_VALIDATOR = instance.getSchema(result.toString("UTF-8"));
-            }
+        try {
+            // load both schemas from resources (root (flags.json) and referenced (targeting.json)
+            // we don't want to resolve anything from the network
+            Map<String, String> mappings = new HashMap<>();
+            mappings.put("https://flagd.dev/schema/v0/targeting.json", "classpath:targeting.json");
+            mappings.put("https://flagd.dev/schema/v0/flags.json", "classpath:flags.json");
+
+            SCHEMA_VALIDATOR = JsonSchemaFactory
+                    .getInstance(SpecVersion.VersionFlag.V7,
+                            builder -> builder
+                                    .schemaMappers(schemaMappers -> schemaMappers.mappings(mappings)))
+                    .getSchema(new URI("https://flagd.dev/schema/v0/flags.json"));
```

Comment:
I think this is simpler, and it supports loading multiple schemas so they can be referenced from a single root.

Reply:
Nice. At first glance I thought this is a network call, but then saw the classpath loading :) This is really good improvement as we don't maintain custom stream loading logic

Reply:
I thought so too.

I also thought a classpath uri might solve the issue of loading classpath flag definitions...

Reply:
>  also thought a classpath uri might solve the issue of loading classpath flag definitions...

The notation is good (prefix `classpath:`)and I like it. Internally it's loading content using the classloader :)